src/HealthChecks.Nats/NatsHealthCheck.cs

```diff
@@ -28,9 +28,25 @@ public Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, Canc
     {
         try
         {
-            _connection ??= CreateConnection(_options);
-            Guard.ThrowIfNull(_connection);
-            var healthCheckResult = GetHealthCheckResultFromState(_connection);
+            // Create new connection if there is no existing one
+            IConnection? connection = _connection;
+            if (connection == null)
+            {
+                connection = CreateConnection(_options);
+                var exchanged = Interlocked.CompareExchange(ref _connection, connection, null);
```

Comment:
Not sure if I'm aware of the complete picture, but if `CompareExchange` is used to throw away connection objects created by other threads maybe double checked locking would be sufficient on the critical section that creates the connection or using `Lazy<T>`

Reply:
Eventually they all do the same. I prefer non-blocking APIs if applicable.